src/modm/platform/dma/stm32/dma_hal.hpp.in

```diff
@@ -0,0 +1,304 @@
+	template <uint32_t id>
+	static constexpr uint32_t getBaseAddress() {
+		if (id == 1)
+			return DMA1_BASE;
+		else
+			return DMA2_BASE;
+	}
+	/**
+	 * Get the base address of the DMA channel reigsters
+	 *
+	 * @tparam id The number of the DMA channel
+	 */
+	template <uint32_t id>
+	static constexpr uint32_t getChannelBaseAddress() {
+		if (id == 1)
+			return DMA1_Channel1_BASE;
+		else
+			return DMA2_Channel1_BASE;
+	}
+	/**
+	 * Get the address of the channel selection register
+	 *
+	 * @tparam id The number of the DMA controller
+	 */
+	template <uint32_t id>
+	static constexpr uint32_t getCselAddress() {
+		if (id == 1)
+			return DMA1_CSELR_BASE;
+		else
+			return DMA2_CSELR_BASE;
+	}
+
+public:
+	/// DMA base register address
+	static constexpr uint32_t DMA_BASE { getBaseAddress<ID>() };
+	/// DMA channel selection register address
+	static constexpr uint32_t DMA_CSEL { getCselAddress<ID>() };
+	/// DMA channel base register address
+	static constexpr uint32_t CHANNEL_BASE { getChannelBaseAddress<ID>() };
+	/// Register offset from channel to channel
+	static constexpr uint32_t CHANNEL_2_CHANNEL { 0x14 };
+
+	static void
+	clearInterruptFlags(uint32_t flags)
+	{
+		DMA_TypeDef *DMA = reinterpret_cast<DMA_TypeDef *>(DMA_BASE);
+		DMA->IFCR |= flags;
+	}
+
+	static uint32_t
+	getInterruptFlags()
+	{
+		DMA_TypeDef *DMA = reinterpret_cast<DMA_TypeDef *>(DMA_BASE);
+		return DMA->ISR;
+	}
+};
+
+/**
+ * Hardware abstraction layer of a DMA channel
+ *
+ * @tparam ChannelID the ID of the channel
+ * @tparam CHANNEL_BASE base address of the channel registers
+ *
+ * @author	Mike Wolfram
+ * @ingroup	modm_platform_dma
+ */
+template <DmaBase::Channel ChannelID, uint32_t CHANNEL_BASE>
+class DmaChannelHal : public DmaBase
+{
+	/**
+	 * Direction of DMA transfer
+	 *
+	 * Initialized to invalid value, must be configured before setting the
+	 * source and destination addresses.
+	 */
+	static inline DmaBase::DataTransferDirection direction {
+		modm::platform::DmaBase::DataTransferDirection(0xff)
+	};
+
+public:
+	/**
+	 * Configure the DMA channel (HAL)
+	 *
+	 * Stops the DMA channel and writes the new values to its control register.
+	 *
+	 * @param[in] dir Direction of the DMA channel
+	 * @param[in] memoryDataSize Size of data in memory (byte, halfword, word)
+	 * @param[in] peripheralDataSize Size of data in peripheral (byte, halfword, word)
+	 * @param[in] memoryIncrement Defines whether the memory address is incremented
+	 * 			  after a transfer completed
+	 * @param[in] peripheralIncrement Defines whether the peripheral address is
+	 * 			  incremented after a transfer completed
+	 * @param[in] priority Priority of the DMA channel
+	 * @param[in] circularMode Transfer data in circular mode?
+	 */
+	static void
+	configure(DataTransferDirection dir, MemoryDataSize memoryDataSize,
+			PeripheralDataSize peripheralDataSize,
+			MemoryIncrementMode memoryIncrement,
+			PeripheralIncrementMode peripheralIncrement, Priority priority,
```

Comment:
```suggestion
			PeripheralIncrementMode peripheralIncrement,
			Priority priority = Priority::Medium,
```
Can we specify a default priority argument (Medium?) here?

Reply:
Done.